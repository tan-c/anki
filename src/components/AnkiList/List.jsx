import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
// import toastr from 'toastr';
import { connect } from 'react-redux';
import { Map } from 'immutable';
import {
  Form, TextArea, Grid, Segment
} from 'semantic-ui-react';

import TextareaControlledConnected from 'utility-react-component/Form/TextareaControlled';
import {
  AnkiActions,
  ankisSortedSelector,
  activeAnkiSelector,
} from 'utility-redux/anki';

import { UiActions } from 'utility-redux/ui';

import AnkiRowConnected from './Row';

export class AnkiList extends React.Component {
  onKeyDownSaveAnki = (event) => {
    const { selectedAnkiTagId } = this.props;
    const question = this.questionInput.ref.value;
    const answer = this.answerInput.ref.value;

    if (event.which === 13 && event.metaKey) {
      const ankiInfo = {
        question,
        answer,
        type: 'anki',
      };

      if (selectedAnkiTagId.length) {
        ankiInfo.tags = [selectedAnkiTagId];
      }

      this.props.AnkiActions.create(ankiInfo).then((res) => {
        this.questionInput.ref.value = '';
        this.answerInput.ref.value = '';
      });
    }
  }

  toggleType = (anki) => {
    this.props.AnkiActions.update(anki.set('type', anki.get('type') === 'code' ? 'anki' : 'code'));
  }

  render() {
    const {
      activeAnki, ankisSorted,
    } = this.props;

    return (
      <React.Fragment>
        <Grid.Row>
          <Grid.Column>
            {activeAnki.size > 0
              && (
                <Form>
                  <TextareaControlledConnected
                    inputName="question"
                    record={activeAnki}
                    actions={this.props.AnkiActions}
                    inputClassNames="width-100p height-60"
                    inputType="textarea"
                    textareaRow={2}
                    callback={(_) => { this.props.UiActions.updateIn(['activeAnkiId'], ''); }}
                  />

                  <TextareaControlledConnected
                    inputName="answer"
                    record={activeAnki}
                    actions={this.props.AnkiActions}
                    inputClassNames="width-100p height-100"
                    inputType="textarea"
                    textareaRow={4}
                    callback={(_) => { this.props.UiActions.updateIn(['activeAnkiId'], ''); }}
                  />
                </Form>
              )
            }

            {activeAnki.size === 0
              && (
                <Form>
                  <Form.Group>
                    <TextArea
                      rows="2"
                      type="text"
                      name="question"
                      ref={(ref) => { this.questionInput = ref; }}
                      placeholder={activeAnki.size ? activeAnki.get('question') : ''}
                    />
                  </Form.Group>
                  <Form.Group>
                    <TextArea
                      rows="4"
                      type="text"
                      name="answer"
                      ref={(ref) => { this.answerInput = ref; }}
                      placeholder={activeAnki.size ? activeAnki.get('answer') : ''}
                      onKeyDown={this.onKeyDownSaveAnki}
                    />
                  </Form.Group>
                </Form>
              )
            }
          </Grid.Column>
        </Grid.Row>

        <Grid.Row>
          <Grid.Column>
            <Grid
              divided="vertically"
              style={{
                overflow: 'auto',
                maxHeight: 'calc(100vh - 300px)'
              }}
            >
              {ankisSorted.valueSeq().map(anki => (
                <AnkiRowConnected
                  anki={anki}
                  key={anki.get('_id')}
                />
              ))}
            </Grid>
          </Grid.Column>
        </Grid.Row>
      </React.Fragment>
    );
  }
}

AnkiList.defaultProps = {
  activeAnki: Map(),
  // activeNote: Map(),
  ankisSorted: Map(),
  selectedAnkiTagId: ''
};

AnkiList.propTypes = {
  // activeNote: PropTypes.object,
  activeAnki: PropTypes.object,
  ankisSorted: PropTypes.object,
  selectedAnkiTagId: PropTypes.string,
  // activeNoteVariousAnkis: PropTypes.object,

  AnkiActions: PropTypes.object.isRequired,
  UiActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    activeAnki: activeAnkiSelector(state),
    // activeNote: activeNoteSelector(state),
    ankisSorted: ankisSortedSelector(state),
    selectedAnkiTagId: state.ui.getIn(['selectedAnkiTagId']),
    // activeNoteVariousAnkis: activeNoteVariousAnkisSelector(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    AnkiActions: bindActionCreators(AnkiActions, dispatch),
    UiActions: bindActionCreators(UiActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(AnkiList);
