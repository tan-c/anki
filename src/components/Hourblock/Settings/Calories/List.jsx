// import React from 'react';
// import PropTypes from 'prop-types';
// import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';

// import { CalorieActions } from 'utility-redux/hourblock/calorie';
// import InputNewConnected from 'utility-react-component/Form/Input/New';
// import Input from 'utility-react-component/Form/Input/Uncontrolled';
// import SelectConnected from 'utility-react-component/Form/Select';

// export class CaloriesList extends React.Component {
//   render() {
//     const { calories } = this.props;

//     return (
//       <React.Fragment>
//         <div className="section-header">
//           Calorie Item List
//         </div>
//         <div className="section-content">
//           <div className="list-with-pinned-bottom">
//             {calories.valueSeq().map(calorie => (
//               <div className="flex-container-row typical-setup" key={calorie.get('_id')}>
//                 <span className="flex-3">
//                   <Input
//                     inputName="name"
//                     record={calorie}
//                     actions={this.props.CalorieActions}
//                   />
//                 </span>
//                 <span className="flex-1">
//                   <SelectConnected
//                     onChangeEvent={(event) => {
//                       this.props.CalorieActions.update(calorie.set('type', event.target.value));
//                     }}
//                     value={calorie.get('type')}
//                     options={['main', 'addition', 'snack', 'drink']}
//                   />
//                 </span>
//                 <span className="flex-1">
//                   <Input
//                     inputName="calorie"
//                     record={calorie}
//                     actions={this.props.CalorieActions}
//                   />
//                 </span>
//               </div>))}
//           </div>
//           <div className="flex-container-row pinned-bottom border-top">
//             <InputNewConnected
//               inputName="name"
//               inputClassNames="flex-1"
//               actions={this.props.CalorieActions}
//             />
//           </div>
//         </div>
//       </React.Fragment>
//     );
//   }
// }

// CaloriesList.defaultProps = {
// };

// CaloriesList.propTypes = {
//   calories: PropTypes.object.isRequired,
//   CalorieActions: PropTypes.object.isRequired,
// };

// function mapStateToProps(state, ownProps) {
//   return {
//     calories: state.calories,
//   };
// }

// function mapDispatchToProps(dispatch) {
//   return {
//     CalorieActions: bindActionCreators(CalorieActions, dispatch),
//   };
// }

// export default connect(mapStateToProps, mapDispatchToProps)(CaloriesList);
