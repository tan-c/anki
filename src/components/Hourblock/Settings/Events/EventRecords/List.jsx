import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map } from 'immutable';
import moment from 'moment';

import { EventRecordActions, selectedEventEventRecordsSelector } from 'utility-redux/hourblock/eventRecord';

export class EventRecordsList extends React.Component {
  render() {
    const { eventRecords, selectedEvent } = this.props;
    return (
      <React.Fragment>
        <div className="section-header">Event Records</div>
        <div className="section-content">
          { eventRecords.valueSeq().map(rec => (
            <div key={rec.get('_id')} className="margin-bottom-10">
              <div className="flex-container-row text-left bg-white-10">
                <span className="flex-1">
                  {moment(rec.get('startedAt')).add(parseInt(rec.get('sectionOfDay'), 10) / 2, 'hour').format('MM-DD HH:mm')}
                </span>
                <i role="button" tabIndex="-1" className="fa fa-fw fa-close" onClick={_ => this.props.EventRecordActions.deleteRecord(rec)} />
              </div>

              {['duration', 'venue', 'spending', 'value', 'comment'].filter(field => selectedEvent.getIn(['details', field])).map(field => (
                <div className="flex-container-row border-bottom-white-20" key={field}>
                  <span className="flex-1 text-right border-right-white-20">{field}</span>
                  <span className="flex-3 text-left">
                    {rec.getIn(['details', field])}
                  </span>
                </div>))}
            </div>))
          }
        </div>
      </React.Fragment>
    );
  }
}

EventRecordsList.defaultProps = {
  eventRecords: Map(),
  selectedEvent: Map(),
};

EventRecordsList.propTypes = {
  eventRecords: PropTypes.object,
  selectedEvent: PropTypes.object,

  EventRecordActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    eventRecords: selectedEventEventRecordsSelector(state),
    selectedEvent: state.events.get(state.ui.getIn(['hourblock', 'settingsPage', 'selectedEventId'])),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    EventRecordActions: bindActionCreators(EventRecordActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(EventRecordsList);
