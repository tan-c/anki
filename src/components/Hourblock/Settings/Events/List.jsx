import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map } from 'immutable';

import { EventActions, eventsSortedByProjectThenCategorySelector } from 'utility-redux/event';
import { UiActions } from 'utility-redux/ui';

import ProjectSelectConnected from 'utility-react-component/Form/HourblockProjectSelect';
import Input from 'utility-react-component/Form/Input/Uncontrolled';
import InputNewConnected from 'utility-react-component/Form/Input/New';

export class EventsList extends React.Component {
  // constructor(props, context) {
  //   super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //   this.setState({
  //   });
  // }

  toggleField = (selectedEventId, field) => {
    const event = this.props.eventsSorted.get(selectedEventId);
    this.props.EventActions.update(event.setIn(['details', field], !event.getIn(['details', field])));
  }

  updateEventProject = (event, rec) => {
    const { value } = event.target;
    let newRecord = rec;
    if (value === '') {
      // Delete wont work, set to null instead
      newRecord = rec.set('project', null);
    } else {
      newRecord = rec.set('project', value);
    }
    this.props.EventActions.update(newRecord);
  }

  render() {
    const { eventsSorted, eventRecords, selectedEventId } = this.props;

    return (
      <React.Fragment>
        <section className="flex-1">
          <div className="section-header">Event List</div>
          <div className="section-content">
            <div className="list-with-pinned-bottom">
              {eventsSorted.valueSeq().map(rec => (
                <div
                  className={`flex-container-row typical-setup ${rec.get('_id') === selectedEventId && 'bg-orange'}`}
                  key={rec.get('_id')}
                >
                  <span className="flex-1">
                    <ProjectSelectConnected
                      onChangeEvent={event => this.updateEventProject(event, rec)}
                      value={rec.getIn(['project', '_id'])}
                      color={rec.getIn(['project', 'category', 'color'])}
                    />
                  </span>

                  <span role="button" tabIndex="-1" className="flex-2" onClick={_ => this.props.UiActions.updateIn(['hourblock', 'settingsPage', 'selectedEventId'], rec.get('_id'))}>
                    <Input
                      inputName="name"
                      record={rec}
                      actions={this.props.EventActions}
                    />
                  </span>

                  <span className="width-20">
                    {eventRecords.valueSeq().filter(record => record.getIn(['event', '_id']) === rec.get('_id')).count() === 0
                      && <i role="button" tabIndex="-1" className="fa fa-fw fa-close" onClick={_ => this.props.EventActions.deleteRecord(rec)} />
                    }
                  </span>
                </div>))
              }
            </div>

            <div className="flex-container-row pinned-bottom border-top">
              <InputNewConnected
                inputName="name"
                actions={this.props.EventActions}
              />
            </div>
          </div>
        </section>

        <section className="flex-1 border-white">
          <div className="section-header">
Details -
            {eventsSorted.getIn([selectedEventId, 'name'])}
          </div>
          {selectedEventId.length && ['duration', 'venue', 'spending', 'value', 'comment'].map(field => (
            <div className="flex-container-row" key={field}>
              <span className="flex-1">{field}</span>
              <i role="button" tabIndex="-1" className={`fa fa-fw fa-check ${eventsSorted.getIn([selectedEventId, 'details', field]) && 'color-green'}`} onClick={_ => this.toggleField(selectedEventId, field)} />
            </div>))}
        </section>
      </React.Fragment>
    );
  }
}

EventsList.defaultProps = {
  eventsSorted: Map(),
  eventRecords: Map(),
  selectedEventId: '',
};

EventsList.propTypes = {
  eventsSorted: PropTypes.object,
  eventRecords: PropTypes.object,
  selectedEventId: PropTypes.string,

  EventActions: PropTypes.object.isRequired,
  UiActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    eventsSorted: eventsSortedByProjectThenCategorySelector(state),
    eventRecords: state.eventRecords,
    selectedEventId: state.ui.getIn(['hourblock', 'settingsPage', 'selectedEventId']),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    EventActions: bindActionCreators(EventActions, dispatch),
    UiActions: bindActionCreators(UiActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(EventsList);
