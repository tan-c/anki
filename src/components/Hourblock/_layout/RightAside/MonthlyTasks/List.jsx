import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import moment from 'moment-timezone';
import { Map } from 'immutable';

import InputNewConnected from 'utility-react-component/Form/Input/New';
import { TaskActions, monthlyTasksSelector } from 'utility-redux/hourblock/task';

export class MonthlyTasksList extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      showMonthlyTask: true,
      showMonthlyTaskRewards: false,
    };
  }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const { monthlyTasks, activeMonth } = this.props;
    const { showMonthlyTask, showMonthlyTaskRewards } = this.state;

    return (
      <div data-role="monthlytasks-list" className="margin-top-10 margin-bottom-10 border-white">
        <span className="flex-container-row typical-setup border-bottom-white-dotted text-left">
          <InputNewConnected
            inputName="content"
            inputClassNames="flex-3"
            placeholderContent="generic montly task..."
            newRecord={{
              targetCompletion: moment().tz('Asia/Tokyo').startOf('year').add(activeMonth, 'months')
                .toDate(),
              type: 'monthly',
            }}
            actions={this.props.TaskActions}
          />

          <i
            role="button" tabIndex="-1"
            className={`width-15 fa fa-fw fa-eye ${showMonthlyTask && 'color-green'}`} onClick={_ => this.setState({
              showMonthlyTask: !showMonthlyTask,
            })}
          />

          <i
            role="button" tabIndex="-1"
            className={`width-15 fa fa-fw fa-gift ${showMonthlyTaskRewards && 'color-green'}`} onClick={_ => this.setState({
              showMonthlyTaskRewards: !showMonthlyTaskRewards,
            })}
          />
        </span>


        { showMonthlyTask && monthlyTasks.has(activeMonth.toString()) && monthlyTasks.get(activeMonth.toString()).sort((a, b) => a.getIn(['project', 'category', 'naturalId']) - b.getIn(['project', 'category', 'naturalId'])).map(task => (
          <React.Fragment key={task.get('_id')}>
            <div className="flex-container-row typical-setup border-bottom-white-20">
              <span className="width-10" style={{ background: task.getIn(['project', 'category', 'color']) }} />
              <span className={`flex-1 text-left ${task.get('isCompleted') && 'color-grey text-line-through'}`}>
                {task.get('content')}
              </span>

              <i
                role="button" tabIndex="-1" className="fa fa-fw fa-check width-15"
                onClick={_ => this.props.TaskActions.deleteRecord(task)}
              />
            </div>

            {showMonthlyTaskRewards && task.has('reward') && (
              <div className="flex-container-row height-20 line-height-20">
                <i className="width-20 fa fa-fw fa-arrow-right line-height-20" />
                <i className="width-20 fa fa-fw fa-gift line-height-20" />
                <span className="flex-1">{task.get('reward')}</span>
              </div>
            )}
          </React.Fragment>
        ))
        }
      </div>
    );
  }
}

MonthlyTasksList.defaultProps = {
  monthlyTasks: Map(),
};

MonthlyTasksList.propTypes = {
  monthlyTasks: PropTypes.object,
  activeMonth: PropTypes.number.isRequired,

  TaskActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    activeMonth: ownProps.activeMonth,
    monthlyTasks: monthlyTasksSelector(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    TaskActions: bindActionCreators(TaskActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(MonthlyTasksList);
