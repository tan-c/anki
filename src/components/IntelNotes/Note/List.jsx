import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Button } from 'semantic-ui-react';

import {
  NoteActions,
  activeNotebookIdSelector,
  activeNotebookNotesSortedSelector
} from 'utility-redux/note';

import NoteRowConnected from './Row';

export class NoteList extends React.Component {
  render() {
    const {
      activeNotebookNotes,
      activeNotebookId
    } = this.props;

    return (
      <React.Fragment>
        <div className="text-center border-bottom-black">
          {`Notes: ${activeNotebookNotes.size} - - `}

          {activeNotebookId.length && (
            <Button
              size="tiny"
              onClick={(_) => {
                this.props.NoteActions.create({
                  notebook: activeNotebookId,
                  title: 'new note'
                });
              }}
            >
              Create New
            </Button>
          )}
        </div>

        {
          activeNotebookNotes.valueSeq().map(note => (
            <NoteRowConnected
              key={note.get('_id')}
              note={note}
            />
          ))
        }
      </React.Fragment>);
  }
}

NoteList.defaultProps = {
  activeNotebookId: '',
};

NoteList.propTypes = {
  activeNotebookId: PropTypes.string,
  activeNotebookNotes: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    activeNotebookId: activeNotebookIdSelector(state),
    activeNotebookNotes: activeNotebookNotesSortedSelector(state),
  };
}


function mapDispatchToProps(dispatch) {
  return {
    NoteActions: bindActionCreators(NoteActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(NoteList);
