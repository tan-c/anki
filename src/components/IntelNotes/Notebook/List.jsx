import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Map } from 'immutable';
// import { bindActionCreators } from 'redux';

import { notebookGroupNotebooksSortedSelector } from 'utility-redux/notebook';
import NotebookListRowConnected from './Row';

export class NotebookList extends React.Component {
  // constructor(props, context) {
  //  super(props, context);
  // }

  // componentWillReceiveProps(nextProps) {
  //  this.setState({
  //   });
  // }

  render() {
    const {
      notebookGroupNotebooks,
    } = this.props;

    return (
      <React.Fragment>
        {notebookGroupNotebooks.valueSeq().map(
          notebook => (
            <NotebookListRowConnected
              key={notebook.get('_id')}
              notebook={notebook}
            />
          )
        )
        }
      </React.Fragment>
    );
  }
}

NotebookList.defaultProps = {
  notebookGroupNotebooks: Map(),
};

NotebookList.propTypes = {
  notebookGroupNotebooks: PropTypes.object,
};

function mapStateToProps(state, ownProps) {
  return {
    notebookGroupNotebooks: notebookGroupNotebooksSortedSelector(state, ownProps.notebookGroupId),
  };
}

export default connect(mapStateToProps)(NotebookList);
