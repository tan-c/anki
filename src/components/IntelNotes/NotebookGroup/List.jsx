import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Map } from 'immutable';

import { NotebookGroupActions } from 'utility-redux/notebookGroup';

import DustbinRecordConnected from './DustbinRecord';
import RecentNoteRowConnected from './RecentNoteRow';
import NotebookListConnected from '../Notebook/List';
import NotebookGroupListRowConnected from './Row';

export class NotebookGroupList extends React.Component {
  // constructor(props) {
  //   super(props);
  //   this.state = {
  //     editMode: 0,
  //   };
  // }

  render() {
    const {
      notebookGroups,
    } = this.props;

    return (
      <div data-role="notebookgroup-list" id="notebook-list">
        <DustbinRecordConnected />
        <RecentNoteRowConnected />
        <div className="spacing-10" />

        {notebookGroups.valueSeq().sort((a, b) => (a.get('title') > b.get('title') ? 1 : -1)).map(notebookGroup => (
          <div className="margin-bottom-10" key={notebookGroup.get('_id')}>
            <NotebookGroupListRowConnected notebookGroup={notebookGroup} />
            <NotebookListConnected notebookGroupId={notebookGroup.get('_id')} />
          </div>))}

        <button
          id="new-notebook-group"
          className="bg-green"
          onClick={(_) => {
            this.props.NotebookGroupActions.create({
              title: 'test-notebook',
            });
          }}
        >
          + Notebook Group
        </button>
      </div>
    );
  }
}

NotebookGroupList.defaultProps = {
  notebookGroups: Map(),
};

NotebookGroupList.propTypes = {
  notebookGroups: PropTypes.object,

  NotebookGroupActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    notebookGroups: state.notebookGroups,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    NotebookGroupActions: bindActionCreators(NotebookGroupActions, dispatch),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(NotebookGroupList);
