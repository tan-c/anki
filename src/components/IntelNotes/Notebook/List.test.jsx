import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

import { Map } from 'immutable';

import { NotebookList } from './List';

function setup() {
  const props = {
    notebookGroupNotebooks: Map(),
  };

  return shallow(<NotebookList {...props} />);
}

describe('NotebookList Unit Tests', () => {
  describe('', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      const jsonOutput = toJson(wrapper);
      // expect(jsonOutput).toMatchSnapshot();
      // expect(wrapper.find("div[data-role='notebook-list']")).toHaveLength(1);
    });
  });
});
