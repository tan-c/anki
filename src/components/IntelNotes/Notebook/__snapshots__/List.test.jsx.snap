// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotebookList Unit Tests  renders correctly 1`] = `<Fragment />`;
